File: Javascript/BubbleSort.js

/**
 * @author Franz Ludeña
 */

function bubbleSort(nums) {

    let size = nums.length;

    // Bucle principal
    for (let i = 0; i < size - 1; i++) {

        // Compara elementos vecinos
        for (let j = 0; j < size - i - 1; j++) {

            // Intercambio de posiciones
            if (nums[j] > nums[j + 1]) {

                [nums[j], nums[j + 1]] = [nums[j + 1], nums[j]];
            }
        }
    }
}

let nums = [5, 1, 6, 2, 4, 3];

bubbleSort(nums);

console.log(nums);

File: Python/BubbleSort.py

# @author Franz Ludeña

def bubble_sort(nums):

    size = len(nums)

    # Bucle principal
    for i in range(size - 1):

        # Compara elementos vecinos
        for j in range(size - i - 1):

            # Intercambio de posiciones
            if nums[j] > nums[j + 1]:
                nums[j], nums[j + 1] = nums[j + 1], nums[j]


nums = [5, 1, 6, 2, 4, 3]

bubble_sort(nums)

print(nums)